Add Clone Functionality.

silverhadch opened the pull request:
Now you can clone an Distrobox and it will Create an Name-Clone Distrobox.

Fixes #25

DenysMb commented:
Looks good. I pushed some changes so we refresh the list of distroboxes now when finishing clone and I changed from "-Clone" to "-clone" because, usually, people write their distroboxes name lowercase.

I don't like that now we have a lot of buttons on the distrobox card. The ideal is that we have there only actions that we use a lot. Clone a distrobox is not a action that we want to be always there (the same for delete).

I think we should have an option on the global drawer called "Clone Container..." that opens a dialog where user can select the container and write the name that they want to the new cloned container. This way we get rid of the button on the card and also gives the ability to the user to give any name they want to the container.

silverhadch commented:
> Looks good. I pushed some changes so we refresh the list of distroboxes now when finishing clone and I changed from "-Clone" to "-clone" because, usually, people write their distroboxes name lowercase.
> 
> I don't like that now we have a lot of buttons on the distrobox card. The ideal is that we have there only actions that we use a lot. Clone a distrobox is not a action that we want to be always there (the same for delete).
> 
> I think we should have an option on the global drawer called "Clone Container..." that opens a dialog where user can select the container and write the name that they want to the new cloned container. This way we get rid of the button on the card and also gives the ability to the user to give any name they want to the container.

If we want to do that then we would need to make the exportedApps Struct more robust. Which i am not against. Will think of a good approach tomorrow.

silverhadch commented:
@DenysMb 
Now all not that necessary toolbar buttons are hidden behind an advanced button. So only package installs, managing exports and unexports and open terminal are visible next to it is advanced that shows the rest of the buttons.

DenysMb commented:
All good. I just changed the label of the view more and clone button and moved the remove button to the end of the list.
I will work on the clone dialog on the develop branch.

silverhadch commented:
> All good. I just changed the label of the view more and clone button and moved the remove button to the end of the list. I will work on the clone dialog on the develop branch.

Well if you do that we should make the exported apps logic check if the .desktop file we loop through only contains the distrobox enter and containername but no -- command then its a distrobox shortcut since for example if one create from Box A A-whatever the exportedApps logic will mistake it for an application because it assumes <containername>-*.desktop as syntax.
